dll/win32/imm32/imm.c

```diff
@@ -1296,6 +1309,73 @@ BOOL WINAPI ImmSetActiveContextConsoleIME(HWND hwnd, BOOL fFlag)
     return ImmSetActiveContext(hwnd, hIMC, fFlag);
 }
 
+/************************************************************************/
+/* STACK_CHECK */
+
+#ifdef STACK_CHECK
+
+DWORD_PTR __security_cookie = 0x0000BB40;
+DWORD_PTR __security_cookie_complement = 0xFFFF44BF;
+
+static VOID Imm32GenerateSecurityCookie(VOID)
```

Comment:
Why are you re-implementing the stack checks the compiler inserts?

Reply:
How can we insert the stack checks into `IMM32`?

Reply:
https://learn.microsoft.com/en-us/archive/msdn-magazine/2017/december/c-visual-c-support-for-stack-based-buffer-protection

Reply:
But all that is automatically done when you compile any module and the CRT is linked to it...
Now, I've tried to look at our compilation settings and I see that by default, `/GS` appears to be disabled (see the msvc.cmake results here:
https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=%2FGS
)
However I don't know whether they get reenabled somehow later.

The `_security_cookie` stuff in the CRT is implemented here:
https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=security_cookie

Reply:
Here is what {{Imm32LoadImeDpi}} looks like on current master (or close to it anyway) in msvc:

![image](https://user-images.githubusercontent.com/610685/201532038-1eb65804-c756-4ecf-9603-d35a7c74844b.png)

![image](https://user-images.githubusercontent.com/610685/201532045-964ff643-26d5-4b60-9faa-bc36b51ef860.png)

Reply:
This is indeed RTC "only". The GuardStack feature (with "security cookies") doesn't seem to be enabled indeed.